dev/tasks/vcpkg-tests/cpp-build-vcpkg.bat

```diff
@@ -0,0 +1,69 @@
+@rem Licensed to the Apache Software Foundation (ASF) under one
+@rem or more contributor license agreements.  See the NOTICE file
+@rem distributed with this work for additional information
+@rem regarding copyright ownership.  The ASF licenses this file
+@rem to you under the Apache License, Version 2.0 (the
+@rem "License"); you may not use this file except in compliance
+@rem with the License.  You may obtain a copy of the License at
+@rem
+@rem   http://www.apache.org/licenses/LICENSE-2.0
+@rem
+@rem Unless required by applicable law or agreed to in writing,
+@rem software distributed under the License is distributed on an
+@rem "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+@rem KIND, either express or implied.  See the License for the
+@rem specific language governing permissions and limitations
+@rem under the License.
+
+@rem Run VsDevCmd.bat to set Visual Studio environment variables for building
+@rem on the command line. This is the path for Visual Studio Enterprise 2019
+call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=amd64
+
+@rem Install build dependencies with vcpkg
+@rem TODO(ianmcook): change --x-manifest-root to --manifest-root after it
+@rem changes in vcpkg
+vcpkg install ^
+    --triplet x64-windows ^
+    --x-manifest-root cpp  ^
+    --clean-after-build ^
+    || exit /B 1
+
+set VCPKG_INSTALLED=%cd%\cpp\vcpkg_installed
+
+@rem Build Arrow C++ library
+mkdir cpp\build
+pushd cpp\build
+
+cmake -G "Visual Studio 16 2019" -A x64 ^
+      -DARROW_BOOST_USE_SHARED=ON ^
+      -DARROW_BUILD_STATIC=OFF ^
+      -DARROW_BUILD_TESTS=ON ^
+      -DARROW_CXXFLAGS="/MP" ^
```

Comment:
How about using `--parallel %NUMBER_OF_PROCESSOR%` with `cmake --build` instead of specifying `/MP` here?

Reply:
Sure, I'll change it to that. FYI this `cmake` command was mostly copied from [ci/appveyor-cpp-build.bat](https://github.com/apache/arrow/blob/master/ci/appveyor-cpp-build.bat#L91-L121) with changes only as needed.

Reply:
Upon further investigation, a few sources point to this `/MP` method having better compatibility with Visual Studio than `--parallel`, so I think I will keep it this way

Reply:
Could you share the sources?

Reply:
- https://gitlab.kitware.com/cmake/cmake/-/issues/20564
- https://blog.kitware.com/cmake-building-with-all-your-cores/ (recommends `/MP`, but it's from 2013)
- https://gitlab.kitware.com/cmake/cmake/-/merge_requests/1962 (seems to show `--parallel` support is somewhat new, and many Windows users run Visual Studio 2017 or newer which I'm not sure supports it)

Reply:
Thanks. I see.
Could you note https://gitlab.kitware.com/cmake/cmake/-/issues/20564 as a comment? We will reconsider this when CMake improves about this.